app/webpacker/components/CompetitionForm/FormSections/DuesEstimate.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { Input, Modal } from 'semantic-ui-react';
+import { QueryClient, useQuery } from '@tanstack/react-query';
+import useInputState from '../../../lib/hooks/useInputState';
+import { calculateDuesUrl } from '../../../lib/requests/routes.js.erb';
+import { currenciesData } from '../../../lib/wca-data.js.erb';
+import Loading from '../../Requests/Loading';
+import Errored from '../../Requests/Errored';
+import { fetchJsonOrError } from '../../../lib/requests/fetchWithAuthenticityToken';
+
+const CALCULATE_DUES_QUERY_CLIENT = new QueryClient();
+
+export default function DuesEstimate({
+  close, countryId, currencyCode, baseEntryFee, competitorLimit,
+}) {
+  const [competitorCount, setCompetitorCount] = useInputState(competitorLimit || 0);
+  const currencyInfo = currenciesData.byIso[currencyCode] || currenciesData.byIso.USD;
+
+  const {
+    data, isLoading, isError,
+  } = useQuery({
+    queryKey: [countryId, currencyCode, baseEntryFee],
```

Comment:
Super nitpick, but you should definitely still include something like `'dues-estimate'` or a similar string literal in the key.

If you only pass in the parameters relevant for `queryFn`, then you won't be able to distinguish _what_ is being fetched. An integer `42` might be a user ID, or a registration ID, or a schedule event ID, or whatever.

In your case, it's not crucial because you're using an isolated `CALCULATE_DUES_QUERY_CLIENT` which doesn't share its memory at all. But if you use an application-wide shared client, then you need the queryKey to be `['user', userId]` or `['registration', registrationId]` instead of just `[userId]`.

Reply:
Okay done, added.